### PR TITLE
Change image.android to use the new image.Cache class from widgets.

### PR DESCRIPTION
**Pull request by hshristov**
Implemented image memory cache (Android).
Implemented image disk cache (Android).
Implemented `loadMode='async'` for Android.
Implemented `decodeWidth`, `decodeHeight` and `useCache` (Android).

### CODE REVIEW

**Comment by vchimev** [comment]
widgets

**Comment by vchimev** [comment]
widgets-uitests-android

**Comment by vchimev** [comment]
widgets-uitests-ios

**Comment by hshristov** [comment]
widgets

**Comment by hshristov** [comment]
widgets-uitests-android

**Comment by hshristov** [comment]
widgets-uitests-ios

**Comment by vchimev** [comment]
run ci

**Comment by vchimev** [comment]
widgets

**Comment by vchimev** [comment]
uitests

**Comment by vchimev** [comment]
run ci

**Comment by vchimev** [comment]
run ci

**Comment by vchimev** [comment]
uitests

**Comment by vchimev** [comment]
run ci

**Comment by vchimev** [comment]
uitests

**Comment by vchimev** [comment]
run ci

**Comment by hshristov** [comment]
run ci

**Comment by hshristov** [comment]
uitests

**Comment by vchimev** [comment]
run ci

**Comment by vchimev** [comment]
uitests

**Comment by dtopuzov** [comment]
run ci

**Comment by enchev** [comment]
run ci

File: tns-core-modules/ui/image/image.android.ts [hunk 1]
```diff
@@ -4,24 +4,27 @@ import proxy = require("ui/core/proxy");
...
-function ensureEnums() {
-    if (!enums) {
-        enums = require("ui/enums");
-    }
-}
+const FILE_PREFIX = "file:///";
+let ASYNC = "async";
+let imageFetcher: org.nativescript.widgets.image.Fetcher;
+let imageCache: org.nativescript.widgets.image.Cache;
+let currentMode: number;
```
Comment (hdeshev) [inline tns-core-modules/ui/image/image.android.ts]: Are `currentMode` values documented somewhere? Can we use an enum instead of a number?

**Comment by hshristov** [comment]
uitests

**Comment by hshristov** [comment]
run ci

**Comment by hshristov** [comment]
uitests

**Pull request merged**